client/sites/hooks/test/use-transfer-with-software-status-query.tsx

```diff
@@ -0,0 +1,53 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { renderHook, waitFor } from '@testing-library/react';
+import nock from 'nock';
+import React from 'react';
+import { useTransferWithSoftwareStatus } from '../use-transfer-with-software-status-query';
+
+// Mock wpcom
+jest.mock( '@automattic/calypso-config', () => ( {
+	isEnabled: jest.fn(),
+} ) );
+
+const mockSuccessResponse = {
+	software_transfer_status: 'pending',
+};
+
+describe( 'useTransferWithSoftwareStatus', () => {
+	beforeAll( () => nock.disableNetConnect() );
+
+	beforeEach( () => nock.cleanAll() );
+
+	afterEach( () => jest.resetAllMocks() );
+
+	it( 'should fetch transfer status successfully', async () => {
+		const queryClient = new QueryClient();
+		const wrapper = ( { children }: { children: React.ReactNode } ) => (
+			<QueryClientProvider client={ queryClient }>{ children }</QueryClientProvider>
+		);
+		nock( 'https://public-api.wordpress.com' )
+			.get( '/wpcom/v2/sites/123/atomic/transfer-with-software/456' )
+			.query( { http_envelope: 1 } )
+			.reply( 200, mockSuccessResponse );
+
+		const { result } = renderHook( () => useTransferWithSoftwareStatus( 123, 456 ), { wrapper } );
+
+		await waitFor( () => expect( result.current.isSuccess ).toBe( true ) );
+		expect( result.current.data ).toEqual( mockSuccessResponse );
+	} );
+
+	it( 'should not fetch when siteId or atomicTransferId is missing', () => {
+		const queryClient = new QueryClient();
+		const wrapper = ( { children }: { children: React.ReactNode } ) => (
+			<QueryClientProvider client={ queryClient }>{ children }</QueryClientProvider>
+		);
+		const { result } = renderHook( () => useTransferWithSoftwareStatus( 0, 456 ), { wrapper } );
+
+		expect( result.current.isFetching ).toBe( false );
+		expect( nock.isDone() ).toBe( true ); // No pending nock requests
+	} );
```

Comment:
Can we add separate test cases for when the `siteId` is missing and when the `atomicTransferId` is missing? We're only testing one of the two cases at present.

Reply:
Whoops! Good catch :D